### DISCUSSION TITLE
A note of thanks and a suggestion ...

### DISCUSSION THREAD

**Discussion by kedarmhaswade** [Digital-Media/HagenbergThesis#166] [General]
This is a monumental work that I found only by chance. Thanks for making these templates available to everyone.

One suggestion I have is regarding using the templates. Instructions ask the user to download a zip/tarball and use the `main.tex` and other files from there. Wouldn't it be better to clone the repo instead so that one can track the upstream and update as needed?

(Sorry if this is already discussed.)

**Comment by imagingbook** [comment]
Thank you - and there is more interesting stuff coming soon!
What you suggest was actually the orginal setup. It was changed to avoid the duplication of `sty` and `cls` files in the various document subdirectories (the style and class files are now kept in a single place of the repository). I.e., when you clone the repository, running `main.tex` inside some document subdirectory (associated with one of the ZIP files) will not work, unless you change your local LaTeX search path too. Also note that cloning the repo incurs considerable overhead (ca 380 MB). Another motive is to streamline processing with Overleaf (by uploading ZIP files), which has become very popular. 
However, working with a repo clone _is_  possible with some extra steps (and of course you should rename `main.tex`).

**Comment by kedarmhaswade** [comment]
Thank you. Yes, it makes sense. Overleaf is pretty popular for a good reason. Thanks for providing the .zip files!
